File: LeetCode/Mathematics/LC-60 Permutation Sequence/LC60.cs
using System;
using System.Collections.Generic;

public class LC60
{
    public static string GetPermutation(int n, int k)
    {
        List<int> list = new List<int>();
        for (int i = 1; i <= n; i++)
        {
            list.Add(i);
        }

        string returnString = "";
        k--;

        while (n > 0)
        {
            int factorialValue = Factorial(n - 1);
            int x = k / factorialValue;
            returnString += list[x];
            list.RemoveAt(x);
            n--;
            k %= factorialValue;
        }

        return returnString;
    }

    public static int Factorial(int n)
    {
        if (n == 1 || n == 0)
            return 1;
        return n * Factorial(n - 1);
    }

    public static void Main(string[] args)
    {
        Console.WriteLine(GetPermutation(3, 3)); // Output: "213"
        Console.WriteLine(GetPermutation(4, 9)); // Output: "2314"
        Console.WriteLine(GetPermutation(3, 1)); // Output: "123"
    }
}

File: LeetCode/Mathematics/LC-60 Permutation Sequence/LC60.java
import java.util.ArrayList;

public class LC60 {
    public static String getPermutation(int n, int k) {
        ArrayList<Integer> list=new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            list.add(i);
        }
        String returnString = "";
        k--;
        while(n>0){
            int factorialValue = factorial(n-1);
            int x = k / factorialValue;
            returnString += list.get(x);
            list.remove(x);
            n--;
            k %= factorialValue;
        }
        return returnString;
    }

    public static int factorial(int n){
        if (n==1 || n==0)
            return 1;
        return n * factorial(n-1);
    }

    public static void main(String[] args){
        System.out.println(getPermutation(3,3));
        System.out.println(getPermutation(4,9));
        System.out.println(getPermutation(3,1));
    }
}
